concept/type_/type_manager.rs

```diff
@@ -617,39 +750,74 @@ impl<Snapshot: WritableSnapshot> TypeManager<Snapshot> {
 
     pub(crate) fn storage_delete_owns_ordering(&self, snapshot: &mut Snapshot, owns_edge: TypeEdge<'_>) {
         debug_assert_eq!(owns_edge.prefix(), Prefix::EdgeOwns);
-        snapshot.delete(
-            build_property_type_edge_ordering(owns_edge).into_storage_key().into_owned_array(),
-        )
+        snapshot.delete(build_property_type_edge_ordering(owns_edge).into_storage_key().into_owned_array())
     }
-    pub(crate) fn storage_set_plays(&self, snapshot: &mut Snapshot, player: impl ObjectTypeAPI<'static>, role: RoleType<'static>) {
+    pub(crate) fn storage_set_plays(
+        &self,
+        snapshot: &mut Snapshot,
+        player: impl ObjectTypeAPI<'static>,
+        role: RoleType<'static>,
+    ) {
         let plays = build_edge_plays(player.clone().into_vertex(), role.clone().into_vertex());
         snapshot.put(plays.into_storage_key().into_owned_array());
         let plays_reverse = build_edge_plays_reverse(role.into_vertex(), player.into_vertex());
         snapshot.put(plays_reverse.into_storage_key().into_owned_array());
     }
 
-    pub(crate) fn storage_delete_plays(&self, snapshot: &mut Snapshot, player: impl ObjectTypeAPI<'static>, role: RoleType<'static>) {
+    pub(crate) fn storage_delete_plays(
+        &self,
+        snapshot: &mut Snapshot,
+        player: impl ObjectTypeAPI<'static>,
+        role: RoleType<'static>,
+    ) {
         let plays = build_edge_plays(player.clone().into_vertex(), role.clone().into_vertex());
         snapshot.delete(plays.into_storage_key().into_owned_array());
         let plays_reverse = build_edge_plays_reverse(role.into_vertex(), player.into_vertex());
         snapshot.delete(plays_reverse.into_storage_key().into_owned_array());
     }
 
-    pub(crate) fn storage_set_relates(&self, snapshot: &mut Snapshot, relation: RelationType<'static>, role: RoleType<'static>) {
+    pub(crate) fn storage_set_plays_overridden(
+        &self,
+        snapshot: &mut Snapshot,
+        plays: Plays<'static>,
+        overridden: Plays<'static>,
+    ) {
+        let property_key =
+            build_property_type_edge_override(plays.into_type_edge()).into_storage_key().into_owned_array();
+        let overridden_plays = ByteArray::copy(overridden.into_type_edge().into_bytes().bytes());
+        snapshot.put_val(property_key, overridden_plays);
+    }
```

Comment:
Do we store the edge or just the role/attribute type?

Reply:
We just need to store the overridden role type (in general, RHS of the `as`) in the property, so what you've done is good